src/libbson/src/bson/bson-json-private.h

```diff
@@ -23,6 +23,7 @@
 struct _bson_json_opts_t {
    bson_json_mode_t mode;
    int32_t max_len;
+   bool is_outermost_array;
```

Comment:
Ensure `is_outermost_array` is initialized (to `true`?) in `bson_json_opts_new()` and add a test ensuring correct backwards-compatible default behavior (use of `bson_json_opts_t` should not require an explicit call to `bson_json_opts_set_outermost_array` for validity).

Reply:
@kevinAlbs said to not change the `bson_json_opts_new()` function as that would be an API break that requires a major release version. This is why we opted to use a "setter" function instead with `bson_json_opts_set_outermost_array()`. I'm not sure if ensuring backwards-compatibility would fix the API break issue, but if it does I can go ahead and initialize the variable in `bson_json_opts_new()`.

Having the variable be uninitialized annoyed me as well but I couldn't find a way in C to have a "default" value in a struct.

Reply:
The _interface_ and corresponding documented behavior must not be changed (e.g. the addition of a new function parameter), but the _implementation_ is free to be changed. Adding a data member to `bson_json_opts_t` and ensuring it is initialized is an implementation detail, not a change to the interface. There is no observable difference to the user; they cannot even observe the addition of a new data member due to `bson_json_opts_t` being opaque. This is the great maintainability benefit of opaque pointers.

The "default" value of data members is mandated by restricting creation of said (opaque) structs to constructors such as `bson_json_opt_new()` that ensure their value is set. There is (by construction) no other way for the user to create the object.

Reply:
> Ensure `is_outermost_array` is initialized (to `true`?) in `bson_json_opts_new()`

Great catch. `bson_json_opts_new` uses `bson_malloc` to allocate storage for the `bson_json_opts_t` struct. This results in `is_outermost_array` being uninitialized. I expect `is_outermost_array` to be initialized to `false` in `bson_json_opts_new()`.

Reply:
Also note, given the few number of data members, initialization can be made into a single statement using C99 compound literals:
```c
opts = (bson_json_opts_t *) bson_malloc (sizeof *opts);
*opts = (bson_json_opts_t){
   .mode = mode,
   .max_len = max_len,
   .is_outermost_array = true,
};
```Recommend using this pattern when able so that the `-Wmissing-field-initializers` warning becomes more effective.

Reply:
Got it, thanks for catching that.